### PR TITLE
mgr/dashboard: Add protractor-screenshoter-plugin

### PR DESCRIPTION
**Pull request by alfonsomthd**
A plugin to save screenshots on E2E tests failures as well as
console logs, raw HTML, ...
All info visualized in an HTML report (screenshots are clickable to see full image).
Useful for debugging/fixing E2E tests.
2019-03-25-protractor-screenshoter-plugin-report

Signed-off-by: Alfonso Martínez <almartin@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/protractor.conf.js [hunk 1]
```diff
@@ -28,6 +28,18 @@ exports.config = {
...
   },
+
+  plugins: [{
+      package: 'protractor-screenshoter-plugin',
+      screenshotPath: '.protractor-report',
+      screenshotOnExpect: 'failure',
+      screenshotOnSpec: 'none',
+      withLogs: true,
+      writeReportFreq: 'asap',
+      imageToAscii: 'none',
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/protractor.conf.js]: omg. we should enable this! :rofl:

**Comment by callithea** [comment]
Would it make sense to add some information about this new e2e test feature to our ``HACKING.rst``?

**Comment by alfonsomthd** [comment]
> Would it make sense to add some information about this new e2e test feature to our `HACKING.rst`?

@callithea Right. Done!

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 2]
```diff
@@ -140,6 +140,12 @@ Start all frontend e2e tests by running::
 
   $ ./run-frontend-e2e-tests.sh
 
+Report:
+  After running tests, you can find the report as well as screenshots of failed test cases
```
Comment (callithea) [inline src/pybind/mgr/dashboard/HACKING.rst]:
```suggestion
  After running the tests you can find the corresponding report as well as screenshots of failed test cases
```

Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/HACKING.rst]: Done.

**Pull request merged**